client-sniffer/build.gradle

```diff
@@ -61,8 +61,11 @@ forbiddenApisTest {
 
 //JarHell is part of es core, which we don't want to pull in
 jarHell.enabled=false
-//NamingConventionCheck is part of test-framework, which we don't want to pull in as it depends on es core
-namingConventions.enabled=false
+
+namingConventions {
+  //we don't have integration tests
+  skipIntegTestInDisguise = true
```

Comment:
tried defaulting it to null, but ended up having to set it in too many projects. Seems like we should rather disable in the client exception only? I went for setting it to null here explicitly but those null values become trickier to handle and understand, I went for the boolean that we already had cause it's more explicit and considered it a bug that when the check is disabled we still load the class.